Project/Source/Components/ComponentTransform.cpp

```diff
@@ -226,10 +207,15 @@ float3 ComponentTransform::GetGlobalPosition() {
 
 Quat ComponentTransform::GetGlobalRotation() {
 	CalculateGlobalMatrix();
-	float4x4 newTransform_ = globalMatrix;
-	newTransform_.Orthogonalize3();
-	newTransform_.Orthonormalize3();
-	return Quat(newTransform_.SubMatrix(3, 3));
+	float3x3 globalRotation = globalMatrix.RotatePart();
```

Comment:
I believe there is a problem here in this method. If an ancestor of a rotated GameObject has a non-uniform scale, the global rotation matrix may not be orthogonal, and although you are normalizing the axes, that does not reorthogonalize the 3x3 matrix. Then, the Quat will not be correctly constructed, leading to an incorrect global rotation (I think it will log a failed assumption).

Maybe I'm wrong, someone should test it. I can test it on sunday, but not before.

Reply:
The global matrix is orthogonalized in the CalculateGlobalMatrix().
`globalMatrix.Orthogonalize3(); // Solution for non-uniform scaled objects`
But I'm testing all this part in detail